src/main/java/org/broadinstitute/consent/http/models/UserRole.java

```diff
@@ -24,29 +27,72 @@
     private Boolean profileCompleted;
 
 
-    public UserRole(){
+    public UserRole() {
     }
 
-    public UserRole(Integer roleId, String name){
+    public UserRole(Integer roleId, String name) {
         this.roleId = roleId;
         this.name = name;
-        this.emailPreference = true;
     }
 
-    public UserRole(Integer roleId, String name, Boolean emailPreference){
+    public UserRole(Integer roleId, String name, String rationale, String status) {
         this.roleId = roleId;
         this.name = name;
-        this.emailPreference = emailPreference;
-    }
-
-    public UserRole(Integer roleId, String name, Boolean emailPreference, String rationale, String status){
-        this.roleId = roleId;
-        this.name = name;
-        this.emailPreference = emailPreference;
         this.rationale = rationale;
         this.status = status;
     }
 
+    /**
+     * Convenience method for backwards compatibility support for older clients.
+     *
+     * @param json A json string that may or may not be correctly structured as a UserRole
+     */
+    public UserRole(String json) {
+        JsonObject jsonObject = new JsonParser().parse(json).getAsJsonObject();
+        Logger log = LoggerFactory.getLogger(UserRole.class.getName());
+        if (jsonObject.has("roleId") && !jsonObject.get("roleId").isJsonNull()) {
+            try {
+                this.setRoleId(jsonObject.get("roleId").getAsInt());
+            } catch (Exception e) {
```

Comment:
Is this idiomatic Gson? I'm used to Jackson and not familiar with this pattern -- wondering if there are particular cases where you'd want to raise an exception instead of logging?

Reply:
Pretty sure this is not idiomatic anything, unfortunately. The goal here is to do JSON conversion in the case where we don't have a json structure that matches the POJO. I think maybe logging a warning here might be better, but I expect that the older (deployed) client code will be triggering that pretty regularly and really shouldn't be an error in that case. The non-standard JSON is picked apart in other places to handle the legacy clients.

Reply:
This can be done a lot more cleanly, thanks for the advice.